Dockerfile

```diff
@@ -3,8 +3,9 @@
 # docs/developer/reproducibility.md.
 
 ARG DEBIAN_IMAGE_DATE=20250224
+ARG DEBIAN_IMAGE_DIGEST=sha256:12c396bd585df7ec21d5679bb6a83d4878bc4415ce926c9e5ea6426d23c60bdc
 
-FROM debian:bookworm-${DEBIAN_IMAGE_DATE}-slim AS dangerzone-image
+FROM debian:bookworm-${DEBIAN_IMAGE_DATE}-slim@${DEBIAN_IMAGE_DIGEST} AS dangerzone-image
```

Comment:
We could also add the registry hostname here, as in `docker.io/library/debian:bookworm-${DEBIAN_IMAGE_DATE}-slim@${DEBIAN_IMAGE_DIGEST}`, if we wanted to remove the choice from the user (building this image with `podman`, for example, prompts the user to choose between `docker.io/library/...` or `quay.io/...`).

Users can set a preference for this in `registries.conf` (`man 5 containers-registries.conf`), but it may be preferable for this repository to specify an upstream (users can always remap the registry to their preferred one anyway).

`docker` (used in this repository's pipelines) will default to `docker.io/library` and prefix the image URI with this, so the workflow files and any docker users will not encounter the prompt.